tests/model_registry/model_catalog/test_model_search.py

```diff
@@ -294,3 +296,228 @@ def test_multiple_artifact_type_filtering(
+        )
+        assert is_valid2, f"API search results do not match database query for '{case_variant}': {errors2}"
+
+        models1 = response1.get("items", [])
+        models2 = response2.get("items", [])
+
+        model_ids1 = sorted([m.get("id") for m in models1])
+        model_ids2 = sorted([m.get("id") for m in models2])
+
+        assert model_ids1 == model_ids2, (
+            f"Case insensitive search failed:\n"
+            f"'{search_term}' returned {len(models1)} models\n"
+            f"'{case_variant}' returned {len(models2)} models"
+        )
+
+    def test_q_parameter_no_results(
+        self: Self,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+        model_registry_namespace: str,
+    ):
+        """Test search with term that should return no results using database validation"""
+        nonexistent_term = "nonexistent_search_term_12345_abcdef"
+        LOGGER.info(f"Testing search for nonexistent term: {nonexistent_term}")
+
+        response = get_models_from_catalog_api(
+            model_catalog_rest_url=model_catalog_rest_url,
+            model_registry_rest_headers=model_registry_rest_headers,
+            q=nonexistent_term,
+        )
+
+        # Validate API results against database query
+        is_valid, errors = validate_search_results_against_database(
+            api_response=response,
+            search_term=nonexistent_term,
+            namespace=model_registry_namespace,
+        )
+        assert is_valid, f"API search results do not match database query for '{nonexistent_term}': {errors}"
+
+        models = response.get("items", [])
+        assert len(models) == 0, f"Expected no results for '{nonexistent_term}', got {len(models)} models"
+
+    @pytest.mark.parametrize("search_term", ["", None])
+    def test_q_parameter_empty_query(
+        self: Self,
+        search_term,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+        model_registry_namespace: str,
+    ):
+        """Test behavior with empty or None q parameter using database validation"""
+        LOGGER.info(f"Testing empty query: {repr(search_term)}")
+
+        response = get_models_from_catalog_api(
+            model_catalog_rest_url=model_catalog_rest_url,
+            model_registry_rest_headers=model_registry_rest_headers,
+            q=search_term,
+        )
+
+        # Only validate against database if search_term is not None/empty
+        # Empty/None queries might have different behavior that doesn't map to our search logic
+        if search_term:
+            is_valid, errors = validate_search_results_against_database(
+                api_response=response,
+                search_term=search_term,
+                namespace=model_registry_namespace,
+            )
+            assert is_valid, f"API search results do not match database query for '{search_term}': {errors}"
```

Comment:
I would even remove, you tested already that the db query has the same results even if it is for a different query. optional